# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../tcp_wrappers/weak_symbols.patch
# Copyright (C) 2006 - 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Inspired from the Fedora Core shared patch without pulling all the shared bits.

  - Rene Rebe <rene@exactcode.de>

--- tcp_wrappers_7.6/Makefile.shared	2003-02-10 20:12:26.000000000 +0100
+++ tcp_wrappers_7.6/Makefile	2003-02-10 20:14:05.000000000 +0100
@@ -391,7 +391,7 @@
 # the ones provided with this source distribution. The environ.c module
 # implements setenv(), getenv(), and putenv().
 
-AUX_OBJ= setenv.o
+AUX_OBJ= weak_symbols.o
 #AUX_OBJ= environ.o
 #AUX_OBJ= environ.o strcasecmp.o
 
@@ -906,5 +913,6 @@
 update.o: mystdarg.h
 update.o: tcpd.h
 vfprintf.o: cflags
+weak_symbols.o: tcpd.h
 workarounds.o: cflags
 workarounds.o: tcpd.h
--- /dev/null	2003-01-30 11:24:37.000000000 +0100
+++ tcp_wrappers_7.6/weak_symbols.c	2003-02-10 20:12:26.000000000 +0100
@@ -0,0 +1,11 @@
+ /*
+  * @(#) weak_symbols.h 1.5 99/12/29 23:50
+  * 
+  * Author: Anthony Towns <ajt@debian.org>
+  */
+
+//#ifdef HAVE_WEAKSYMS
+#include <syslog.h>
+int deny_severity = LOG_WARNING;
+int allow_severity = SEVERITY; 
+//#endif
--- tcp_wrappers_7.6/tcpd.h	1996-03-19 16:22:25.000000000 +0100
+++ ./tcpd.h	2006-10-11 13:10:49.000000000 +0200
@@ -78,8 +78,8 @@
 
 /* Global variables. */
 
-extern int allow_severity;		/* for connection logging */
-extern int deny_severity;		/* for connection logging */
+extern int allow_severity __attribute__ ((weak)); /* for connection logging */
+extern int deny_severity __attribute__ ((weak)); /* for connection logging */
 extern char *hosts_allow_table;		/* for verification mode redirection */
 extern char *hosts_deny_table;		/* for verification mode redirection */
 extern int hosts_access_verbose;	/* for verbose matching mode */
